### PR TITLE
Add GPS, Compass and IR Ball sensors

### PR DESCRIPTION
**Pull request by Adman**
* Ball now emits IR signal
* Robots do not get exact position of other robots and the ball

Signed-off-by: Adrian Matejov <ado.matejov@gmail.com>

- [x] Update script for generating the world

### CODE REVIEW

File: controllers/rcj_soccer_team_blue/rcj_soccer_robot.py [hunk 1]
```diff
@@ -129,42 +118,59 @@ def send_data_to_team(self, robot_id) -> None:
...
-        if robot_ball_angle > 360:
-            robot_ball_angle -= 360
+        # subtract math.pi/2 (90) so that the heading is 0 facing 'north'
+        # (given x going from left to right)
+        rad = math.atan2(compass_values[0], compass_values[2]) - (math.pi / 2)
+        if rad < -math.pi:
+            rad = rad + (2 * math.pi)
 
-        return robot_ball_angle, robot_angle
+        # This world is in NUE (not in EUN), so -1* has to be applied
```
[the remark this answers is not in the document]
Comment (mrshu) [reply controllers/rcj_soccer_team_blue/rcj_soccer_robot.py:173]: Thanks @Adman !

File: controllers/rcj_soccer_team_blue/utils.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
...
 
     Args:
-        ball_angle (float): Angle between the ball and the robot
+        ball_direction (list of floats): Current direction of the ball
 
     Returns:
         int: 0 = forward, -1 = right, 1 = left
     """
-    if ball_angle >= 345 or ball_angle <= 15:
+    if -0.13 <= ball_direction[0] <= 0.13:
```
[the remark this answers is not in the document]
Comment (mrshu) [reply controllers/rcj_soccer_team_blue/utils.py:10]: @Adman I think it would -- thanks!

**Comment by Adman** [comment]
@mrshu All comments should be addressed now.

**Comment by mrshu** [comment]
Thanks @Adman !

**Pull request merged**